swift/ql/lib/codeql/swift/elements/type/Type.qll

```diff
@@ -11,4 +13,22 @@ class Type extends Generated::Type {
    * ```
    */
   Type getUnderlyingType() { result = this }
+
+  /**
+   * Gets any base type of this type, or the result of resolving a typedef. For
+   * example in the following code, `C` has base type `B` which has underlying
+   * type `A`. Thus, `getABaseOrAliasedType*` can be used to discover the
+   * relationship between `C` and `A`.
+   * ```
+   * class A {}
+   *
+   * typealias B = A
+   *
+   * class C : B {}
+   * ```
+   */
+  Type getABaseOrAliasedType() {
+    result = this.(NominalType).getABaseType() or
+    result = this.(TypeAliasType).getAliasedType()
+  }
```

Comment:
So the problem we're trying to solve with this predicate is that, we want to do a transitive closure over "getABaseType", but since `getABaseType`'s root definition is on `NominalType` we "stop early" when we hit a `TypeAliasType`. Is that correct?

Would it not be better to handle this more "automatically" by:
1. Moving the definition of `getABaseType` to the `Type` class, and
2. Use `getAliasedType().getABaseType()` (which is now available because we moved `getABaseType` to `Type`) when `this` is a `TypeAliasType`?

Reply:
Done.  Doing the same for `TypeDecl` is going to be a bit more involved (and irrelevant to this PR) so I'll do that as follow-up.